components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/cdc/CDCProcessor.java

```diff
@@ -0,0 +1,238 @@
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_SCHEMA_HISTORY_INTERNAL;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_SCHEMA_HISTORY_INTERNAL_FILE_FILENAME;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_TOPIC_PREFIX;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_VALUE_CONVERTER;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_VALUE_CONVERTER_SCHEMAS_ENABLE;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.DEBEZIUM_SKIPPED_OPERATIONS;
+import static org.wso2.carbon.inbound.endpoint.protocol.cdc.InboundCDCConstants.TRUE;
+
+public class CDCProcessor extends InboundRequestProcessorImpl implements TaskStartupObserver, InboundTaskProcessor {
+
+    private CDCPollingConsumer pollingConsumer;
+    private Properties cdcProperties;
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private static final String ENDPOINT_POSTFIX = "CDC" + COMMON_ENDPOINT_POSTFIX;
+    private static final String FILE_OFFSET_STORAGE_CLASS = "org.apache.kafka.connect.storage.FileOffsetBackingStore";
+    private static final String FILE_SCHEMA_HISTORY_STORAGE_CLASS = "io.debezium.storage.file.history.FileSchemaHistory";
+    private static final Log LOGGER = LogFactory.getLog(CDCProcessor.class);
+
+    private enum operations {create, update, delete, truncate};
+    private enum opCodes {c, u, d, t};
+
+    public CDCProcessor(InboundProcessorParams params) {
+        this.name = params.getName();
+        this.injectingSeq = params.getInjectingSeq();
+        this.onErrorSeq = params.getOnErrorSeq();
+        this.synapseEnvironment = params.getSynapseEnvironment();
+        this.cdcProperties = params.getProperties();
+        setProperties();
+        try {
+            this.interval = Long.parseLong(cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_INTERVAL));
+        } catch (NumberFormatException nfe) {
+            throw new SynapseException("Invalid numeric value for interval.", nfe);
+        }
+        this.sequential = true;
+        if (cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL) != null) {
+            this.sequential = Boolean
+                    .parseBoolean(cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL));
+        }
+        this.coordination = true;
+        if (cdcProperties.getProperty(PollingConstants.INBOUND_COORDINATION) != null) {
+            this.coordination = Boolean.parseBoolean(cdcProperties.getProperty(PollingConstants.INBOUND_COORDINATION));
+        }
+    }
+
+    private void setProperties () {
+        LOGGER.info("Initializing the properties");
```

Comment:
Do we need this log here? If we are adding a log, better to mention that this is related to CDC or something.

Reply:
it was added to help the user to have an idea on, at what point an issue was happened  in case of an error flow. Updated the log as mentioned.